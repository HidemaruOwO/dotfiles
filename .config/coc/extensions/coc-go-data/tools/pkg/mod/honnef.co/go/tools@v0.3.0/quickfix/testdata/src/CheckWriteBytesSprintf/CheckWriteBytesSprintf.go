package pkg

import (
	"bytes"
	"fmt"
	"io"
)

type NotAWriter struct{}

func (NotAWriter) Write(b []byte) {}

func fn1() {
	var w io.Writer
	var w2 NotAWriter

	w.Write([]byte(fmt.Sprint("abc", "de")))   // want `Use fmt.Fprint`
	w.Write([]byte(fmt.Sprintf("%T", w)))      // want `Use fmt.Fprintf`
	w.Write([]byte(fmt.Sprintln("abc", "de"))) // want `Use fmt.Fprintln`

	w2.Write([]byte(fmt.Sprint("abc", "de")))
}

func fn2() {
	buf := new(bytes.Buffer)
	var sw io.StringWriter

	buf.WriteString(fmt.Sprint("abc", "de"))   // want `Use fmt.Fprint`
	buf.WriteString(fmt.Sprintf("%T", 0))      // want `Use fmt.Fprintf`
	buf.WriteString(fmt.Sprintln("abc", "de")) // want `Use fmt.Fprintln`

	// We can't suggest fmt.Fprint here. We don't know if sw implements io.Writer.
	sw.WriteString(fmt.Sprint("abc", "de"))
	sw.WriteString(fmt.Sprintf("%T", 0))
	sw.WriteString(fmt.Sprintln("abc", "de"))
}
