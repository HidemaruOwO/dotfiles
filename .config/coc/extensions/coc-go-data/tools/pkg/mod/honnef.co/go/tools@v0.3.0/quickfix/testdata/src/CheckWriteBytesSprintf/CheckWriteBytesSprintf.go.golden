package pkg

import (
	"bytes"
	"fmt"
	"io"
)

type NotAWriter struct{}

func (NotAWriter) Write(b []byte) {}

func fn1() {
	var w io.Writer
	var w2 NotAWriter

	fmt.Fprint(w, "abc", "de")   // want `Use fmt.Fprint`
	fmt.Fprintf(w, "%T", w)      // want `Use fmt.Fprintf`
	fmt.Fprintln(w, "abc", "de") // want `Use fmt.Fprintln`

	w2.Write([]byte(fmt.Sprint("abc", "de")))
}

func fn2() {
	buf := new(bytes.Buffer)
	var sw io.StringWriter

	fmt.Fprint(buf, "abc", "de")   // want `Use fmt.Fprint`
	fmt.Fprintf(buf, "%T", 0)      // want `Use fmt.Fprintf`
	fmt.Fprintln(buf, "abc", "de") // want `Use fmt.Fprintln`

	// We can't suggest fmt.Fprint here. We don't know if sw implements io.Writer.
	sw.WriteString(fmt.Sprint("abc", "de"))
	sw.WriteString(fmt.Sprintf("%T", 0))
	sw.WriteString(fmt.Sprintln("abc", "de"))
}
